tests/testthat/test-sample_and_adjust.R

```diff
@@ -76,6 +76,78 @@ test_that("trial by dose change works", {
   expect_true(abs((tmp$y - 15)/15) < 0.05)
 })
 
+test_that("ltbs ruv transformation works", {
+  attr(mod, "ltbs") <- TRUE
+  out_ltbs_v1 <- sample_and_adjust_by_dose(
+    regimen_update_design = create_regimen_update_design(
+      at = c(2, 4),
+      anchor = "dose"
+    ),
+    sampling_design = create_sampling_design(
+      offset = c(19, 20, 21, 22, 11, 12, 13, 14),
+      when = rep("dose", 8),
+      at = c(1, 1, 1, 1, 3, 3, 3, 3),
+      anchor = "dose"
+    ),
+    regimen = regimen,
+    pars_true_i = generate_iiv(mod, omega, par, seed = 1),
+    sim_model = mod,
+    sim_ruv = list(prop = 0, add = 0.1),
+    est_model = mod,
+    parameters = par,
+    omega = omega,
+    ruv = list(prop = 0, add = 0.1),
+    target = create_target_design(
+      targettype = "conc",
+      targetvalue = 15,
+      at = 5,
+      anchor = "dose"
+    ),
+    dose_optimization_method = map_adjust_dose
+  )
+
+
+  attr(mod, "ltbs") <- NULL
+  out_ltbs_v2 <- sample_and_adjust_by_dose(
+    regimen_update_design = create_regimen_update_design(
+      at = c(2, 4),
+      anchor = "dose"
+    ),
+    sampling_design = create_sampling_design(
+      offset = c(19, 20, 21, 22, 11, 12, 13, 14),
+      when = rep("dose", 8),
+      at = c(1, 1, 1, 1, 3, 3, 3, 3),
+      anchor = "dose"
+    ),
+    regimen = regimen,
+    pars_true_i = generate_iiv(mod, omega, par, seed = 1),
+    sim_model = mod,
+    sim_ruv = list(prop = 0.1, add = 0),
+    est_model = mod,
+    parameters = par,
+    omega = omega,
+    ruv = list(prop = 0.1, add = 0),
+    target = create_target_design(
+      targettype = "conc",
+      targetvalue = 15,
+      at = 5,
+      anchor = "dose"
+    ),
+    dose_optimization_method = map_adjust_dose
+  )
+
+  mean_pcte_v1 <- mean(
+    (out_ltbs_v1$tdms$y-out_ltbs_v1$tdms$true_y) / out_ltbs_v1$tdms$true_y
+    )
+
+  mean_pcte_v2 <- mean(
+    (out_ltbs_v1$tdms$y - out_ltbs_v1$tdms$true_y) / out_ltbs_v1$tdms$true_y
+  )
+
+  # expect same when ltbs add error and non-ltbs prop error are equal
+  expect_true(mean_pcte_v1 == mean_pcte_v2)
```

Comment:
will this be true with random number generation?

maybe there is a better thing you can test?